### PR TITLE
chore: add device to train configuration

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Should go in after https://github.com/instructlab/instructlab/pull/2460

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [x] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

File: src/instructlab/configuration.py [hunk 1]
```diff
@@ -392,6 +392,12 @@ class _train(BaseModel):
         default=DEFAULTS.MODEL_REPO,
         description="Directory where the model to be trained is stored.",
     )
+    device: str = Field(
+        default="cpu",
+        description="PyTorch device to use. Use 'cpu' for 'simple' and 'full' training. Use 'cuda' for NVidia CUDA / AMD ROCm GPUs. Use 'hpu' for Intel Gaudi GPUs. Use 'mps' for MacOS Metal Performance Shader.",
```
Comment (cdoern) [inline src/instructlab/configuration.py:397]:
```suggestion
        description="PyTorch device to use. Use 'cpu' for 'simple' and 'full' training. Use 'cuda' for Nvidia CUDA / AMD ROCm GPUs. Use 'hpu' for Intel Gaudi GPUs. Use 'mps' for MacOS Metal Performance Shader when running 'full training on a Mac.",
```

File: tests/testdata/default_config.yaml [hunk 2]
```diff
@@ -265,6 +265,10 @@ train:
   # Allow CPU offload for deepspeed optimizer.
   # Default: False
   deepspeed_cpu_offload_optimizer: false
+  # PyTorch device to use. Use 'cpu' for 'simple' and 'full' training.
+  # Use 'cuda' for NVidia CUDA / AMD ROCm GPUs. Use 'hpu' for Intel Gaudi GPUs.
```
Comment (cdoern) [inline tests/testdata/default_config.yaml:269]:
```suggestion
  # Use 'cuda' for Nvidia CUDA / AMD ROCm GPUs. Use 'hpu' for Intel Gaudi GPUs.
```

just run make regenerate-testdata again once resolving the above

**Review by cdoern — COMMENTED** [review]
one comment on the help text

File: src/instructlab/configuration.py (hunk 1 above)
Comment (nathan-weinberg) [reply src/instructlab/configuration.py:397]: Done, made some additional clarifications as well

**Review by cdoern — APPROVED** [review]

**Review by alinaryan — APPROVED** [review]

**Pull request merged**